src/curves/curve.ts

```diff
@@ -24,7 +26,19 @@ export type Curve = {
   publicAdd: (publicKey: Uint8Array, tweak: Uint8Array) => Uint8Array;
   compressPublicKey: (publicKey: Uint8Array) => Uint8Array;
   decompressPublicKey: (publicKey: Uint8Array) => Uint8Array;
-};
+  privateKeyLength: number;
+} & (
+  | {
+      rootKeyPairIsCreatedFrom: 'seed';
+    }
+  | {
+      rootKeyPairIsCreatedFrom: 'entropy';
```

Comment:
Making the least possible amount of changes to existing logic was my to priority, in that sense, having `rootKeyPairIsCreatedFrom` as a property of the curve seems as the most straightforward option. I would agree that conceptually, this attribute should be `deriver` specific, e.g. CIP3 or SLIP10 should have this property, but only `curve` is passed to bip39 `deriveChildKey` so in order to keep that and make the changes simple I kept the logic around `curve`

Reply:
IMHO the name `rootKeyPairIsCreatedFrom` is misleading because technically a "seed" is also obtained from the mnemonic in the end (in practice)

Instead, I'd consider calling this property `masterKeyGenerationSpec` and it could be either `slip10` or `cip3-icarus`

and you could link here the docs
https://github.com/satoshilabs/slips/blob/133ea52a8e43d338b98be208907e144277e44c0e/slip-0010.md#master-key-generation

https://cips.cardano.org/cips/cip3/icarus.md